wagtail/admin/panels.py

```diff
@@ -1114,6 +1114,21 @@ def get_edit_handler(cls):
 Page.get_edit_handler = get_edit_handler
 
 
+@functools.lru_cache(maxsize=0)
+def get_panel(cls):
+    """
+    Get the panel to use in the Wagtail admin when editing this model.
+    """
+    if hasattr(cls, "edit_handler"):
```

Comment:
@gasman do you think we should rename this to ``panel``? (note we already have ``panels`` but this doesn't let you customise the ``ObjectList``)

Reply:
FYI the `extract_panel_definitions_from_model_class` that's called inside this function already does that:

https://github.com/wagtail/wagtail/blob/a316408928c7eb36c3522f70e8799f2284c3d600/wagtail/admin/panels.py#L77-L79

I kept this fallback for `get_snippet_edit_handler`, and I think if we were to remove this fallback, we should also remove it here at the same time:

https://github.com/wagtail/wagtail/blob/a316408928c7eb36c3522f70e8799f2284c3d600/wagtail/admin/panels.py#L1089-L1090

Reply:
> @gasman do you think we should rename this to `panel`? (note we already have `panels` but this doesn't let you customise the `ObjectList`)

I'm vaguely in favour of renaming it, but I don't think `panel` is the right name, even if that would be logically consistent with the other renames - firstly, it's a pretty generic word, which means that it's fairly likely that someone's already using it as a field name; and secondly, it's a long way away from a developer's intuitive understanding of what a panel is. Most developers probably think of panels as a list rather than a tree, so the concept that that the entire editing interface corresponds to one (root) panel is quite alien to them. We could make this more explicit by naming it something like `root_panel`, but I'd really prefer a name that makes immediate sense without having to get up to speed on that background detail. Maybe something like `admin_base_layout`? At that point I'm not sure if anything is a real improvement over `edit_handler`, and maybe we should simply reinvent the term "edit handler" to mean "the top-level panel that handles the entire editing interface for a given model" and avoid the rename after all...

Reply:
This makes sense to me, thanks!